src/main/scala/cromwell/binding/WdlBinding.scala

```diff
@@ -420,42 +428,48 @@ case class WdlBinding(ast: Ast, source: WdlSource, importResolver: ImportResolve
           }
 
           /* All MemberAccess ASTs that are not contained in other MemberAccess ASTs */
-          val memberAccesses = WdlBinding.findAstsWithTrail(inputKv._2.ast, "MemberAccess") filter {
+          val memberAccesses = WdlBinding.findTopLevelMemberAccesses(inputKv._2.ast)
+          /*val memberAccesses = WdlBinding.findAstsWithTrail(inputKv._2.ast, "MemberAccess") filter {
             case (k, v) => !(v map {case a:Ast => a.getName} contains "MemberAccess")
-          }
-          memberAccesses.keys map validateMemberAccess
+          }*/
+          memberAccesses.map(partialEvalMemberAccess).map{y => y.get}
         }
       }
     }
   }
 
   /* Partially evaluate MemberAccess ASTs to make sure they're not nonsense at compile time */
-  def validateMemberAccess(ast: Ast): Any = {
+  def partialEvalMemberAccess(ast: Ast): Try[Call] = {
+    def callFromName(name: String): Try[Call] = {
+      val callList = workflows.head.calls.filter {call =>
+        call.name == name
+      }
+      //val all = bindings ++ callList
+      if (callList.size == 0) {
+        Failure(new SyntaxError(wdlSyntaxErrorFormatter.undefinedMemberAccess(ast)))
+      } else {
+        Success(callList.head)
+      }
+    }
     val rhs = sourceString(ast.getAttribute("rhs"))
     val lhs = ast.getAttribute("lhs") match {
-      case a: Ast => validateMemberAccess(a)
-      case terminal: Terminal =>
-        val name = sourceString(terminal)
-        val bindings = importedBindings filter {b => b.namespace == Some(name)}
-        val taskList = tasks filter {task => task.name == name}
-        val all = bindings ++ taskList
-        if (all.size == 0) throw new SyntaxError(wdlSyntaxErrorFormatter.undefinedMemberAccess(ast))
-        all.head
+      case a: Ast => callFromName(WdlExpression.toString(a))
+      case terminal: Terminal => callFromName(sourceString(terminal))
+        //val name = sourceString(terminal)
+        //val bindings = importedBindings.filter {b => b.namespace == Some(name)}
+        //val callList = workflows.head.calls.filter {call => call.name == name}
+        //val all = bindings ++ callList
+        //if (callList.size == 0) throw new SyntaxError(wdlSyntaxErrorFormatter.undefinedMemberAccess(ast))
+        //Success(callList.head)
     }
 
     lhs match {
-      case b:WdlBinding =>
-        val all = b.executables find {
-          case w:Workflow => w.name == rhs
-          case t:Task => t.name == rhs
-        }
-        if (all.size == 0) throw new SyntaxError(wdlSyntaxErrorFormatter.undefinedMemberAccess(ast))
-        all.head
-      case t:Task =>
-        val wdlValue = t.outputs find{p => p.name == rhs}
-        wdlValue getOrElse {
+      case Success(c:Call) =>
+        c.task.outputs.find{p => p.name == rhs}.getOrElse {
```

Comment:
put a space after the `find` or don't use the infix. also you should be albe to replace the `p => p` with a `_`